src/Magnum/Octree/Octree.h

```diff
@@ -0,0 +1,274 @@
+#ifndef Magnum_Octree_Octree_h
+#define Magnum_Octree_Octree_h
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2015 Andrea Capobianco <andrea.c.2205@gmail.com>
+    Copyright © 2015 Jonathan Hale <squareys@googlemail.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+ * @brief Class @ref Magnum::Octree::Octree
+ *
+ * @author Andrea Capobianco
+ */
+
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Range.h>
+#include <Magnum/Math/Vector3.h>
+#include <Magnum/Math/Vector4.h>
+#include <Magnum/Shapes/AxisAlignedBox.h>
+
+#include <vector>
+#include "visibility.h"
+
+namespace Magnum { namespace Math { namespace Geometry { namespace Intersection {
+
+/**
+ * @brief Which half space of the plane the point is in.
+ * @param point the point
+ * @param plane the plane
+ * @returns a value `> 0` if on positive half space,
+ *                  `< 0` if on negative half space,
+ *                  `0` if on the plane.
+ */
+template<class T>
+float pointPlaneHalfSpace(Vector3<T> point, Vector4<T> plane) {

```

Comment:
No newline after the `template<>` specifier, please. Similarly below.

Also, can you pass the `Vector`s as `const&`?